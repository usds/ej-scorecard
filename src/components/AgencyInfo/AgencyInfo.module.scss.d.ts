declare namespace AgencyInfoNamespace {
  export interface IAgencyInfoScss {
    agencyInfoCol1: string;
    agencyInfoCol2: string;
    agencyInfoLogo: string;
  }
}

declare const AgencyInfoScssModule: AgencyInfoNamespace.IAgencyInfoScss & {
  /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
  locals: AgencyInfoNamespace.IAgencyInfoScss;
};

export = AgencyInfoScssModule;
