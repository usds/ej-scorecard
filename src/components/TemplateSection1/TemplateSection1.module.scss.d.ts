declare namespace TemplateSection1Namespace {
  export interface ITemplateSection1Scss {
    templateSection1Container: string;
  }
}

declare const TemplateSection1ScssModule: TemplateSection1Namespace.ITemplateSection1Scss & {
  /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
  locals: TemplateSection1Namespace.ITemplateSection1Scss;
};

export = TemplateSection1ScssModule;
