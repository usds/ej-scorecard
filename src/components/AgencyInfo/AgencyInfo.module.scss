@use "uswds-core" as *;

.agencyInfoCol1{
    display: flex;
    flex-direction: column;
    align-items: left;
    
};

.agencyInfoCol2{
    display: flex;
    flex-direction: column;
    align-items: flex-start;

    p span {
        display: block;
    }
};

.agencyInfoLogo {
    @include u-margin-left(5);
    @include u-margin-right(5);
    @include u-margin-top(2);
}

.bullet1 {
    @include u-margin-left(4);
    
    li {
        @include u-margin-bottom(2)
    }
}

h1, h2, h3, h4, h5 {
    margin-bottom: unset;
}