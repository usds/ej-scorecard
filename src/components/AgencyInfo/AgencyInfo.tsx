import React from 'react';
import { StaticImage } from 'gatsby-plugin-image';
import { Grid } from '@trussworks/react-uswds';

// import { useAgencyInfo } from '@/hooks/use-agency-info';

import * as styles from './AgencyInfo.module.scss';
import { AgencyInfoProps } from '@/types';
import ScorecardSideNav from '../ScorecardSideNav';

const AgencyInfo: React.FC<AgencyInfoProps> = ({
  info,
  allAgencyNames,
  pathname,
}) => {
  // Todo: may not need this custom hook
  // const agencyInfo = useAgencyInfo();

  return (
    <Grid row gap>
      <Grid desktop={{ col: 8 }} tablet={{ col: 10 }} col={12}>
        <Grid row gap>
          <Grid col={4}>
            <div className={styles.agencyInfoCol1}>
              <StaticImage
                className={styles.agencyInfoLogo}
                src="../../../static/images/agency-a.png"
                alt="Agency logo"
              />
              <h3>{info.Name}</h3>
              <span>{info.Contact_Name}</span>
              <span>{info.Address_Line_1}</span>
              <span>{info.Address_Line_2}</span>
              <span>{info.Phone}</span>
              <span>{info.Site}</span>
            </div>
          </Grid>
          <Grid col={8}>
            <div className={styles.agencyInfoCol2}>
              <h2>{`About ${info.Name}`}</h2>
              <p>{info.About_description}</p>
            </div>
          </Grid>
        </Grid>
      </Grid>
      <Grid desktop={{ col: 1 }} tablet={{ col: 10 }} col={12} />
      <Grid desktop={{ col: 3 }} tablet={{ col: 10 }} col={12}>
        <ScorecardSideNav allAgencyNames={allAgencyNames} pathname={pathname} />
      </Grid>
    </Grid>
  );
};

export default AgencyInfo;
