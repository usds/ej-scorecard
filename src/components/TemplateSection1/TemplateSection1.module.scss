@use "uswds-core" as *;

.templateSection1Container{

}
