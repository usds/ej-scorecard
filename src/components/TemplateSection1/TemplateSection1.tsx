import React from 'react';
import {
  Grid,
  SummaryBox,
  SummaryBoxContent,
  SummaryBoxHeading,
} from '@trussworks/react-uswds';
// import * as styles from './TemplateSection1.module.scss';

import { TemplateSectionProps } from '@/types';
import { DELIMITERS } from '@/data/constants';
import { possessivePlural, splitTextIntoParagraphs } from '../util';

const TemplateSection1: React.FC<TemplateSectionProps> = ({ agencyData }) => {
  const {
    A_EMAIL,
    A_ACRONYM,
    THE_A_NAME,
    J40_TCP,
    J40_ANNOUNCE,
    J40_AMOUNT,
    J40_MOD1,
    J40_MOD2,
    J40_MOD3,
    J40_E1,
    J40_E2,
    J40_E3,
    J40_HIGH,
  } = agencyData;

  return (
    <>
      <Grid row>
        <Grid desktop={{ col: 8 }} tablet={{ col: 10 }} col={12}>
          <h2 className={`scorecard-header2`}>Justice40 Initiative</h2>
        </Grid>
      </Grid>
      <Grid row gap={6}>
        <Grid desktop={{ col: 8 }} tablet={{ col: 10 }} col={12}>
          <p className={`margin-top-0`}>
            As part of President Biden’s
            {` `}
            <a
              className="usa-link usa-link--alt usa-link--external"
              href={`https://www.whitehouse.gov/environmentaljustice/justice40/`}
              target="_blank"
              rel="noreferrer"
            >
              Justice40 Initiative
            </a>
            {` `}, the federal government is working toward the goal that 40
            percent of the overall benefits of certain federal investments reach
            disadvantaged communities that are marginalized and overburdened by
            pollution and underinvestment. In particular, the President’s
            Justice40 Initiative applies to federal investments that address
            climate change, clean energy and energy efficiency, clean transit,
            affordable and sustainable housing, training and workforce
            development, remediation and reduction of legacy pollution, and the
            development of critical clean water and wastewater infrastructure.
          </p>

          <p>
            This Phase One Scorecard provides an update on initial progress made
            by <strong>{THE_A_NAME}</strong> in implementing the Justice40
            Initiative. Future versions of the Environmental Justice Scorecard
            will provide additional information and updates on the benefits of
            Justice40 covered programs.
          </p>

          {A_EMAIL && (
            <p>
              For more information on {agencyData.A_ACRONYM}’s Justice40 covered
              programs, please contact
              {` `}
              <a href={`mailto:${A_EMAIL}`} target="_blank" rel="noreferrer">
                {A_EMAIL}
              </a>
              {` `}.
            </p>
          )}

          <div>
            Phase One Scorecard metrics and highlights in Fiscal Year 2022
            include:
          </div>

          <ul className={`bullet1`}>
            {J40_TCP && (
              <li>
                <strong>{J40_TCP}</strong> Justice40 covered program(s)
              </li>
            )}
            {J40_ANNOUNCE && (
              <li>
                <strong>{J40_ANNOUNCE}</strong>
                {` `}
                funding announcement(s) covered under the Justice40 Initiative
              </li>
            )}
            {J40_AMOUNT && (
              <li>
                <strong>{J40_AMOUNT}</strong> in funding made available from
                Justice40 covered programs
                <sup>1</sup>
              </li>
            )}
            <ul>
              {J40_E1 && <li>{J40_E1}</li>}
              {J40_E2 && <li>{J40_E2}</li>}
              {J40_E3 && <li>{J40_E3}</li>}
            </ul>
            {(J40_MOD1 || J40_MOD2 || J40_MOD3) && (
              <>
                <li>
                  Made program modifications for Justice40 covered programs such
                  as:
                </li>
                <ul>
                  {J40_MOD1 && <li>{J40_MOD1}</li>}
                  {J40_MOD2 && <li>{J40_MOD2}</li>}
                  {J40_MOD3 && <li>{J40_MOD3}</li>}
                </ul>
              </>
            )}
          </ul>

          {J40_AMOUNT && (
            <p>
              <sup>1</sup> This funding amount describes examples of the funding
              that was made available in Fiscal Year 2022 through Justice40
              covered programs. The amount does not represent the total funding
              that benefits disadvantaged communities. The funding was made
              available through grants, notices of funding opportunity, and
              other funding announcements. These funding announcements were from
              Justice40 covered programs that were funded through discretionary
              and mandatory appropriations, as well as supplemental
              appropriations, such as those enacted under the Bipartisan
              Infrastructure Law (Pub. L. 117-58). The Justice40 Initiative sets
              a goal that 40 percent of overall benefits from certain federal
              investments flow to disadvantaged communities.
            </p>
          )}
        </Grid>
        <Grid desktop={{ col: 4 }} tablet={{ col: 10 }} col={12}>
          {J40_HIGH && (
            <SummaryBox>
              <SummaryBoxHeading headingLevel="h3">
                {`${possessivePlural(A_ACRONYM)} highlights`}
              </SummaryBoxHeading>
              <SummaryBoxContent>
                {agencyData.J40_HIGH.includes(DELIMITERS.PARAGRAPH)
                  ? splitTextIntoParagraphs(
                      agencyData.J40_HIGH,
                      DELIMITERS.PARAGRAPH,
                    ).map((paragraph) => paragraph)
                  : agencyData.J40_HIGH}
              </SummaryBoxContent>
            </SummaryBox>
          )}
        </Grid>
      </Grid>
    </>
  );
};

export default TemplateSection1;
