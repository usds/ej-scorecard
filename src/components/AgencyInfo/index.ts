import AgencyInfo from './AgencyInfo';
export default AgencyInfo;
