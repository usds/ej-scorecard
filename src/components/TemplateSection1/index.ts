import TemplateSection1 from './TemplateSection1';
export default TemplateSection1;
