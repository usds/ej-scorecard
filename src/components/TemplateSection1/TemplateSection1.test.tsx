import createComponentWithIntl from '@/test/testHelpers';
import { agencyData } from '../AgencyInfo/AgencyInfo.test';
import TemplateSection1 from './TemplateSection1';

describe(`rendering of TemplateSection1 Component`, () => {
  it(`checks if component renders`, () => {
    const component = createComponentWithIntl(
      <TemplateSection1 agencyData={agencyData} />,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
