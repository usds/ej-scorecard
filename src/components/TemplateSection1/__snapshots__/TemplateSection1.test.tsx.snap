// Vitest Snapshot v1

exports[`rendering of AgencyInfo Component > checks if component renders group 1 1`] = `
<div
  className="grid-row grid-gap"
  data-testid="grid"
>
  <div
    className="grid-col-12 tablet:grid-col-10 desktop:grid-col-8"
    data-testid="grid"
  >
    <div
      className="grid-row grid-gap"
      data-testid="grid"
    >
      <div
        className="grid-col-4"
        data-testid="grid"
      >
        <div>
          <div
            className="gatsby-image-wrapper gatsby-image-wrapper-constrained"
            data-gatsby-image-wrapper=""
            style={
              {
                "backgroundColor": undefined,
                "display": "inline-block",
                "overflow": "hidden",
                "position": "relative",
                "verticalAlign": "top",
              }
            }
          >
            <div
              style={
                {
                  "display": "block",
                  "maxWidth": 334,
                }
              }
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;charset=utf-8,%3Csvg%20height='278'%20width='334'%20xmlns='http://www.w3.org/2000/svg'%20version='1.1'%3E%3C/svg%3E"
                style={
                  {
                    "display": "block",
                    "maxWidth": "100%",
                    "position": "static",
                  }
                }
              />
            </div>
            <div
              aria-hidden={true}
              data-placeholder-image=""
              style={
                {
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
            />
            <picture>
              <source
                data-srcset="/static/703cdc54fb8f1bddfe59854183c3829b/59647/agency-a.webp 84w,/static/703cdc54fb8f1bddfe59854183c3829b/02be2/agency-a.webp 167w,/static/703cdc54fb8f1bddfe59854183c3829b/7adb0/agency-a.webp 334w"
                sizes="(min-width: 334px) 334px, 100vw"
                type="image/webp"
              />
              <img
                alt="Agency logo"
                data-gatsby-image-ssr=""
                data-main-image=""
                data-src="/static/703cdc54fb8f1bddfe59854183c3829b/dff88/agency-a.png"
                data-srcset="/static/703cdc54fb8f1bddfe59854183c3829b/c3bfd/agency-a.png 84w,/static/703cdc54fb8f1bddfe59854183c3829b/53761/agency-a.png 167w,/static/703cdc54fb8f1bddfe59854183c3829b/dff88/agency-a.png 334w"
                decoding="async"
                loading="lazy"
                sizes="(min-width: 334px) 334px, 100vw"
                style={
                  {
                    "backgroundColor": undefined,
                    "height": "100%",
                    "left": 0,
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "opacity": 0,
                    "position": "absolute",
                    "top": 0,
                    "transform": "translateZ(0)",
                    "transition": "opacity 250ms linear",
                    "width": "100%",
                    "willChange": "opacity",
                  }
                }
              />
            </picture>
            <noscript>
              <picture>
                <source
                  sizes="(min-width: 334px) 334px, 100vw"
                  srcSet="/static/703cdc54fb8f1bddfe59854183c3829b/59647/agency-a.webp 84w,/static/703cdc54fb8f1bddfe59854183c3829b/02be2/agency-a.webp 167w,/static/703cdc54fb8f1bddfe59854183c3829b/7adb0/agency-a.webp 334w"
                  type="image/webp"
                />
                <img
                  alt="Agency logo"
                  data-gatsby-image-ssr=""
                  data-main-image=""
                  decoding="async"
                  loading="lazy"
                  sizes="(min-width: 334px) 334px, 100vw"
                  src="/static/703cdc54fb8f1bddfe59854183c3829b/dff88/agency-a.png"
                  srcSet="/static/703cdc54fb8f1bddfe59854183c3829b/c3bfd/agency-a.png 84w,/static/703cdc54fb8f1bddfe59854183c3829b/53761/agency-a.png 167w,/static/703cdc54fb8f1bddfe59854183c3829b/dff88/agency-a.png 334w"
                  style={
                    {
                      "backgroundColor": undefined,
                      "height": "100%",
                      "left": 0,
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "opacity": 0,
                      "position": "absolute",
                      "top": 0,
                      "transform": "translateZ(0)",
                      "transition": "opacity 250ms linear",
                      "width": "100%",
                      "willChange": "opacity",
                    }
                  }
                />
              </picture>
            </noscript>
            <script
              dangerouslySetInnerHTML={
                {
                  "__html": "const t=\\"undefined\\"!=typeof HTMLImageElement&&\\"loading\\"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll(\\"img[data-main-image]\\");for(let e of t){e.dataset.src&&(e.setAttribute(\\"src\\",e.dataset.src),e.removeAttribute(\\"data-src\\")),e.dataset.srcset&&(e.setAttribute(\\"srcset\\",e.dataset.srcset),e.removeAttribute(\\"data-srcset\\"));const t=e.parentNode.querySelectorAll(\\"source[data-srcset]\\");for(let e of t)e.setAttribute(\\"srcset\\",e.dataset.srcset),e.removeAttribute(\\"data-srcset\\");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector(\\"[data-placeholder-image]\\").style.opacity=0)}}",
                }
              }
              type="module"
            />
          </div>
          <h3>
            Test Agency
          </h3>
          <span>
            1 Pennsylvania Avenue N.W.
          </span>
          <span>
            Washington, D.C. 20004
          </span>
          <span>
            (212) 533-3333
          </span>
          <span>
            <a
              href="mailto:"
            />
          </span>
          <span>
            <a
              className="usa-link usa-link--alt usa-link--external"
              href="https://someagency.gov"
              rel="noopener noreferrer"
              target="_blank"
            >
              someagency.gov
            </a>
          </span>
        </div>
      </div>
      <div
        className="grid-col-8"
        data-testid="grid"
      >
        <div>
          <h2>
            About Test Agency
          </h2>
          <p>
            mission
          </p>
           
        </div>
      </div>
    </div>
    <div
      className="grid-row grid-gap"
      data-testid="grid"
    >
      <p>
        This page provides information on how
         
        <strong>
          Test Agency
        </strong>
         is working to reduce climate and environmental burdens and harms in communities across America.
      </p>
      <p>
        
            This information is part of the Federal government’s first Environmental Justice Scorecard, which will be updated and improved each year based on public input and feedback. The Environmental Justice Scorecard tracks the work of Federal agencies to deliver on the Justice 40 Initiative, to implement and enforce environmental and civil rights protections on behalf of communities, and to evaluate and develop agency capacity and resources to institutionalize environmental justice.
            
      </p>
    </div>
  </div>
  <div
    className="grid-col-12 tablet:grid-col-10 desktop:grid-col-1"
    data-testid="grid"
  />
  <div
    className="grid-col-12 tablet:grid-col-10 desktop:grid-col-3"
    data-testid="grid"
  />
</div>
`;

exports[`rendering of ScorecardSideNav Component > checks if component renders 1`] = `null`;

exports[`rendering of TemplateSection1 Component > checks if component renders 1`] = `
[
  <div
    className="grid-row"
    data-testid="grid"
  >
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-8"
      data-testid="grid"
    >
      <h2
        className="scorecard-header2"
      >
        Justice40 Initiative
      </h2>
    </div>
  </div>,
  <div
    className="grid-row grid-gap-6"
    data-testid="grid"
  >
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-8"
      data-testid="grid"
    >
      <p
        className="margin-top-0"
      >
        For the first time in our nation’s history, the Federal government has made it a goal that 40 percent of the overall benefits of certain Federal investments flow to
         
        <a
          className="usa-link usa-link--alt usa-link--external"
          href="https://screeningtool.geoplatform.gov/en/#3/33.47/-97.5"
          rel="noreferrer"
          target="_blank"
        >
          disadvantaged communities
        </a>
         
        that are marginalized, underserved, and overburdened by pollution. The following reflects the historic resources available to communities in Fiscal Year 2022 to implement the Justice40 Initiative. Please visit this page for more information about the
         
        <a
          className="usa-link usa-link--alt usa-link--external"
          href="https://ceq.sites.eop.gov/ceqej/Shared%20Documents/Scorecard/Template/whitehouse.gov/environmentaljustice/justice40"
          rel="noreferrer"
          target="_blank"
        >
          Justice40 Initiative
        </a>
         
        and visit
         
        <a
          className="usa-link usa-link--alt usa-link--external"
          href="aaa.gov/test"
          rel="noreferrer"
          target="_blank"
        >
          aaa.gov/test
        </a>
        .
      </p>
      <p>
        
          The goal of 40 percent of overall benefits refers to all covered programs under the Justice40 Initiative. Information about how these programs will identify and track benefits will be shared in the Phase Two Environmental Justice Scorecard.
          
      </p>
      <ul
        className="bullet1"
      >
        <li>
          <strong>
            100
          </strong>
           Justice40 covered programs
        </li>
        <li>
          <strong>
            100
          </strong>
           
          announcements in Fiscal Year 2022 representing a total of
           
          <strong>
            100
          </strong>
           funding that is available to benefit communities **add footnote, that the funding amounts include annual discretionary and mandatory appropriations, Inflation Reduction Act, and Bipartisan Infrastructure Law funding
        </li>
        <ul>
          <li>
            E1
          </li>
          <li>
            E2
          </li>
          <li>
            E3
          </li>
        </ul>
        <li>
          Made program modifications for Justice40 covered programs such as:
        </li>
        <ul>
          <li />
          <li />
          <li />
        </ul>
      </ul>
    </div>
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-4"
      data-testid="grid"
    >
      <div
        className="usa-summary-box"
        data-testid="summary-box"
      >
        <div
          className="usa-summary-box__body"
        >
          <h3
            className="usa-summary-box__heading"
          >
            Test Agency Justice40 highlights
          </h3>
          <div
            className="usa-summary-box__text"
          >
            Highlight
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
